src/shared/sam/cli/samCliInfo.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { getLogger, Logger } from '../../logger'
+import { ChildProcessResult } from '../../utilities/childProcess'
+import { DefaultSamCliProcessInvoker } from './samCliInvoker'
+import { SamCliProcessInvoker } from './samCliInvokerUtils'
+
+/**
+ * Maps out the response text from the sam cli command `sam --info`
+ */
+export interface SamCliInfoResponse {
+    version: string
+}
+
+export class SamCliInfoInvocation {
```

Comment:
In a separate PR, we should replace all of these invocation classes with a single service, i.e.

```typescript
export interface SamCli {
    build(args: SamCliBuildArguments): Promise<SamCliBuildResult>
    info(args: SamCliInfoArguments): Promise<SamCliInfoResult>
    ...
}
export class DefaultSamCli implements SamCli {
    // Interface implementation
    // Common private methods to validate a `ChildProcessResult`, etc.
}
```

Reply:
https://github.com/aws/aws-toolkit-vscode/issues/558